diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b27de3..76162d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,7 @@ include(./cmake/add_boost.cmake)
 add_boost(TRY_BOOST_VERSION BOOST_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED
           BOOST_NOT_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED ${PROJECT_NAME})
 
-find_package(Threads)
+find_package(Threads REQUIRED)
 
 find_package(OpenSSL REQUIRED)
 
@@ -133,7 +133,7 @@ packageproject(
   CPACK
   YES
   DEPENDENCIES
-  "${ADDBOOSTCMAKE_PACKAGEPROJECT_INSTALL_TARGETS};OpenSSL 3.0")
+  "${ADDBOOSTCMAKE_PACKAGEPROJECT_INSTALL_TARGETS};OpenSSL 3.0;Threads")
 
 if(${${PROJECT_NAME}_BUILD_EXAMPLES})
   add_subdirectory(examples)
