diff --git a/tests/eigen_test/CMakeLists.txt b/tests/eigen_test/CMakeLists.txt
index ac81f83..7cf1f73 100644
--- a/tests/eigen_test/CMakeLists.txt
+++ b/tests/eigen_test/CMakeLists.txt
@@ -2,7 +2,7 @@ project(eigen_test)
 
 add_executable(${PROJECT_NAME} ${PROJECT_NAME}.cpp)
 
-find_package (Eigen3 3.4 QUIET)
+find_package (Eigen3 3.4 REQUIRED)
 if(NOT Eigen3_FOUND)
   FetchContent_Declare(
     Eigen3
@@ -16,9 +16,9 @@ if(NOT Eigen3_FOUND)
   set(EIGEN_BUILD_DOC OFF CACHE BOOL "Don't build Eigen documentation")
   set(EIGEN_BUILD_PKGCONFIG OFF CACHE BOOL "Don't build pkg-config files")
   set(EIGEN_BUILD_TESTING OFF CACHE BOOL "Don't build Eigen tests")
-  
+
   FetchContent_MakeAvailable(Eigen3)
-  
+
   if(NOT TARGET Eigen3::Eigen)
     add_library(Eigen3 INTERFACE IMPORTED GLOBAL)
     add_library(Eigen3::Eigen ALIAS Eigen3)
