diff --git a/test/transitive_dependency/CMakeLists.txt b/test/transitive_dependency/CMakeLists.txt
index 3e17cc9..25b2f1f 100644
--- a/test/transitive_dependency/CMakeLists.txt
+++ b/test/transitive_dependency/CMakeLists.txt
@@ -29,7 +29,7 @@ set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
 
 add_library(${PROJECT_NAME} SHARED source/transitive_dependency.cpp)
 target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
-target_link_libraries(${PROJECT_NAME} PUBLIC fmt::fmt-header-only cxxopts)
+target_link_libraries(${PROJECT_NAME} PUBLIC fmt::fmt-header-only cxxopts::cxxopts)
 
 target_include_directories(
   transitive_dependency PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
