diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7759cfa..5fd1b6e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -352,3 +352,22 @@ if(KLEIDIAI_BUILD_BENCHMARK)
     target_include_directories(kleidiai_benchmark
         PRIVATE KLEIDIAI_BENCHMARK_INCLUDE_DIR)
 endif()
+
+
+find_package(PackageProject.cmake)
+
+packageProject(
+  NAME kleidiai
+  VERSION ${PROJECT_VERSION}
+  NAMESPACE kleidiai
+  BINARY_DIR ${PROJECT_BINARY_DIR}
+  INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include
+  INCLUDE_DESTINATION include/${PROJECT_NAME}
+  VERSION_HEADER "${VERSION_HEADER_LOCATION}" EXPORT_HEADER "${EXPORT_HEADER_LOCATION}"
+  COMPATIBILITY "AnyNewerVersion" DISABLE_VERSION_SUFFIX ON
+  # cmake-format:off DO *NOT* divide this like to different lines. It will destroy a lot of stuff
+  # related to making this project not only installable, but usable via `find_package(the_project)`
+  DEPENDENCIES
+    ""
+  # cmake-format:on
+)
