diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 7a158d6..d068b89 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -96,7 +96,8 @@ if (NOT WIN32)
     llama_target_and_test(test-grammar-parser.cpp)
     llama_target_and_test(test-grammar-integration.cpp)
     llama_target_and_test(test-llama-grammar.cpp)
-    llama_target_and_test(test-chat.cpp)
+    file(GLOB_RECURSE templates CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/../models/templates/*.jinja")
+    llama_target_and_test(test-chat.cpp NAME test-chat ARGS ${templates})
     # TODO: disabled on loongarch64 because the ggml-ci node lacks Python 3.8
     if (NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "loongarch64")
         llama_target_and_test(test-json-schema-to-grammar.cpp   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/..)
