diff --git a/CMakeLists.txt b/CMakeLists.txt
index a91e0e0..8eb517e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,8 @@ add_library(corral INTERFACE
 
 target_include_directories(corral INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
+if(BUILD_TESTING)
+
 set(CORRAL_TEST_RUNNER "" CACHE STRING "Use this wrapper for running tests")
 
 set(
@@ -108,6 +110,7 @@ if(NOT ("${CORRAL_BOOST}" STREQUAL ""))
     add_test(NAME corral_asio_test COMMAND ${CORRAL_TEST_RUNNER} corral_asio_test)
 endif()
 
+endif()
 
 #
 # Installation
